hnn_core/network.py

```diff
@@ -412,6 +414,11 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
+    def __eq__(self, other):
+        if not isinstance(other, Network):
+            return NotImplemented
+        # Check for all attributes (Discuss)
```

Comment:
I think the right way to implement this is to first implement `__eq__` for `Section` and `Cell` object and call them from here. Python builtins already have it implemented so presumably `net.external_drives` etc should be straightforward to check for equality